src/Elm/Kernel/Utils.js

```diff
@@ -37,6 +37,14 @@ function _Utils_eqHelp(x, y, depth, stack)
 		return false;
 	}
 
+        // Handle Json.Encode.Value
+        if (x.$ === 0) {
```

Comment:
I think you have a missing closed bracket.

Also, more importantly, this is not a good enough distinction for JSON. A lot of data (custom types notably) has `{ $: 0, a: ... }` (with or without additional fields).

I'm not saying that this problem should be fixed, but the [docs do say not to compare JSON directly](https://package.elm-lang.org/packages/elm/core/latest/Basics#==), and this is not the way to fix the issue.

Additionally, `x.a == y.a` is not a correct/sufficient check, since it doesn't compare JSON recursively. Meaning that `Json.Encode.string "abc" == Json.Encode.string "abc"` would return True, but `Json.Decode.list Json.Encode.string ["abc"] == Json.Decode.list Json.Encode.string ["abc"]` for instance would return False.

You would need a deep equal function written for JS objects, with all the problems and inconsistencies that they may have (do you compare prototypes, do you compare functions, etc?)